### PR TITLE
Feature/acl simulator

### PR DESCRIPTION
**Pull request by JasonMarechal25**

### CODE REVIEW

File: src/cpp/lpnamer/problem_modifier/LinkProblemsGenerator.cpp [hunk 1]
```diff
@@ -19,77 +25,69 @@
...
+  for (const ActiveLink &link : _links) {
+    for (const Candidate &candidate : link.getCandidates()) {
+      if (problem_modifier.has_candidate_col_id(candidate.get_name())) {
+        std::lock_guard guard(coupling_mutex_);
+        couplings[{candidate.get_name(), problem_name}] =
+            problem_modifier.get_candidate_col_id(candidate.get_name());
+      }
+    }
+  }
+  writer->Write_problem(in_prblm);
```
Comment (a-zakir) [inline src/cpp/lpnamer/problem_modifier/LinkProblemsGenerator.cpp]:
```suggestion
treat( problem_name, couplings, writer, in_prblm, variable_provider);
```

File: src/cpp/lpnamer/problem_modifier/AntaresProblemToXpansionProblemTranslator.cpp [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    const std::string& solver_name) {
+  SolverFactory factory;
+  auto problem = std::make_shared<Problem>(factory.create_solver(solver_name));
+  problem->init();
+  const auto& constant = lps._constant;
+  const auto& hebdo = lps._hebdo.at({year, week});
+
+  std::vector<int> tmp(constant->NombreDeVariables, 0);
+  std::vector<char> coltypes(constant->NombreDeVariables, 'C');
+  // solver_addcols(problem, hebdo->CoutLineaire, tmp, {}, {}, hebdo->Xmin,
```
Comment (a-zakir) [inline src/cpp/lpnamer/problem_modifier/AntaresProblemToXpansionProblemTranslator.cpp]: remove commented code

File: src/cpp/lpnamer/main/RunProblemGeneration.cpp [hunk 3]
```diff
@@ -92,15 +100,139 @@ void RunProblemGeneration(
...
+                  problems_and_data.end(), [&](const auto& problem_and_data) {
+                    const auto& [problem, data] = problem_and_data;
+
+                    auto problem_variables_from_zip_adapter =
+                        std::make_shared<ProblemVariablesZipAdapter>(
+                            reader, data, links, logger);
+                    linkProblemsGenerator.treat(
+                        data._problem_mps, couplings, problem_writer, problem,
+                        problem_variables_from_zip_adapter);
+                  });
```
Comment (a-zakir) [inline src/cpp/lpnamer/main/RunProblemGeneration.cpp:151]: it is possible to merge the "for" and "for_each" loops

File: src/cpp/lpnamer/main/RunProblemGeneration.cpp [hunk 4]
```diff
@@ -92,15 +100,139 @@ void RunProblemGeneration(
...
+  bool use_zip_implementation = true;
+  bool use_file_implementation = false;
+  if (use_zip_implementation) {
+    std::shared_ptr<ArchiveReader> reader =
+        InstantiateZipReader(antares_archive_path);
+    std::shared_ptr<ArchiveWriter> writer;
+    const std::filesystem::path tmpArchivePath = OpenTmpArchive(lpDir_, writer);
+
+    /* Zip problem writer */
+    auto problem_writer =
```
Comment (a-zakir) [inline src/cpp/lpnamer/main/RunProblemGeneration.cpp:122]: writing mps in zip after PGen will be removed see Pr [628](https://github.com/AntaresSimulatorTeam/antares-xpansion/pull/628) and [622](https://github.com/AntaresSimulatorTeam/antares-xpansion/pull/622)

**Review by a-zakir — CHANGES_REQUESTED** [review]

File: src/cpp/lpnamer/main/RunProblemGeneration.cpp (hunk 3 above)
Comment (JasonMarechal25) [reply src/cpp/lpnamer/main/RunProblemGeneration.cpp:151]: Not exactly.
I need the ascociation of "problem" and "data" both contained in two different vectors: xpansion_problems and mpsList.
You are right that I could iterate through both vectors at the same time using one for loop with the same index.
However my goal with the for_each loop is to parallelize the operations and I can't itarate both vector at once with a for_each loop nor use a index based loop.

File: src/cpp/lpnamer/problem_modifier/LinkProblemsGenerator.cpp (hunk 1 above)
Comment (JasonMarechal25) [reply src/cpp/lpnamer/problem_modifier/LinkProblemsGenerator.cpp]: Thank you. I commited a variant using std::move on some values

**Review by a-zakir — APPROVED** [review]

**Pull request merged**